python/lsst/ap/verify/ingestion.py

```diff
@@ -307,12 +344,21 @@ def _doIngestDefects(self, repo, calibRepo, defectTarball):
         defectTarball : `str`
             The name of a .tar.gz file that contains all the compressed
             defect images.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if ``defectTarball`` exists but is empty.
         """
         # TODO: clean up implementation after DM-5467 resolved
         defectDir = os.path.join(calibRepo, "defects")
-        if not os.path.isdir(defectDir):
-            os.mkdir(defectDir)
-        tarfile.open(defectTarball, "r").extractall(defectDir)
+        with tarfile.open(defectTarball, "r") as opened:
+            if opened.getNames():
+                if not os.path.isdir(defectDir):
+                    os.mkdir(defectDir)
```

Comment:
Maybe use pathlib here in place of `if exist: mkdir`, to avoid `os.mkdir` race conditions on parallel ingest:

`pathlib.Path(defectDir).mkdir(parents=True, exist_ok=True) `

https://stackoverflow.com/a/14364249
https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir

Reply:
I'm not sure parallel ingestion is possible, but your solution is more elegant.